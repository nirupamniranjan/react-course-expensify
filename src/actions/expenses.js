
import uuid from 'uuid'
import database from '../firebase/firebase'
import { add } from 'numeral';
export const addExpense = (expense) =>({
    type: 'ADD_EXPENSE',
    expense 

});

export const startAddExpense = (expenseData={description :'',
note:'',
amount:0,
createdAt:0})=>{

    return (dispatch)=>{
        const {description ='',
        note='',
        amount=0,
        createdAt=0}=expenseData;
        const expense={description,note,amount,createdAt};

        
       const submitData=database.ref('expenses').push(expenseData);
        
       return submitData.then((ref)=>{
          
            dispatch(addExpense({
                id:ref.key,
                ...expense
            }))
        }).catch((e)=>{
            console.log(e + 'error occured')
        })
    }
}
export const removeExpense=({id} ={})=>({
    type: 'REMOVE_EXPENSE',
     id 
});

//EDIT Expense

export const editExpense =(id,updates) =>({
    type:'EDIT_EXPENSE',
    id,
    updates
});

export const setExpenses = (expenses) =>({
    type : 'SET_EXPENSES',
    expenses
});

export const startSetExpenses = ()=>{

    return (dispatch)=>{
        
      return database.ref('expenses').once('value').then((snapshot)=>{
       const expenses = [];
       snapshot.forEach((childSnapshot)=>{

        expenses.push({
            id : childSnapshot.key,
            ...childSnapshot.val()
       })
    })
        
        dispatch(setExpenses(expenses))
    
    
    })
        
       
    }
}
