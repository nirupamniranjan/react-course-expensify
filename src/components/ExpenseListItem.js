import React from 'react'
import {Link} from 'react-router-dom'


const expensetListItem=(props)=>(


    <div>
        <p>
            Description:{props.expense.description} {" "}

            amount : {props.expense.amount} {" "}
            createdAt : {props.expense.createdAt}  {" "}
            <Link to={`/edit/${props.expense.id}`}>Edit</Link>
        </p>
    </div>
    

);



export default expensetListItem;