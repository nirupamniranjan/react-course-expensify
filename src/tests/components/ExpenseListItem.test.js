import ExpenseListItem from '../../components/ExpenseListItem'
import React from 'react'
import {shallow} from 'enzyme'
import expenses from'../Fixtures/expenses'

console.log(expenses[0])

test('Test Expense List Item',()=>{
  const wrapper = shallow(<ExpenseListItem expense={expenses[0]}/>)

    expect(wrapper).toMatchSnapshot();

});
