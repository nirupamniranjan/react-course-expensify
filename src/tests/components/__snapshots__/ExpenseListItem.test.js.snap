// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Expense List Item 1`] = `
<div>
  <p>
    Description:
    beer
     
     
    amount : 
    125
     
     
    createdAt : 
    0
      
     
    <Link
      to="/edit/1"
    >
      Edit
    </Link>
  </p>
</div>
`;
